src/store/ChartState.js

```diff
@@ -97,14 +97,24 @@ class ChartState {
 
         if (symbol !== this.symbol) {
             this.symbol = symbol;
+            isSymbolChanged = true;
+
             if (this.mainStore.chart && this.mainStore.chart.feed) {
-                this.mainStore.chart.feed.onMasterDataUpdate(this.scrollChartToLeft);
+                this.mainStore.chart.feed.onMasterDataReinitialize(this.scrollChartToLeft);
             }
         }
 
+        if (onExportLayout !== this.onExportLayout) {
+            this.onExportLayout = onExportLayout;
+            this.exportLayout();
+        }
+
+
         if (chartType !== this.chartType && this.context) {
             this.setChartType(chartType);
-        } else if (granularity !== undefined && granularity !== this.granularity) {
+        }
+
+        if (granularity !== undefined && granularity !== this.granularity) {
```

Comment:
any reason changing to `if` instead of `else if`?

Reply:
@kelly-binary yes, because it should be possible to change chartType and granularity at the same time.